source/Graphite.System/AppPoolListener.cs

```diff
@@ -11,65 +11,84 @@ namespace Graphite.System
     internal class AppPoolListener
     {
         private readonly string appPoolName;
+        private readonly string category = "Process";
+        private readonly string counter = "Working Set";
 
         private string counterName;
 
-        private CounterListener workingSetListener;
+        private CounterListener counterListener;
+        private bool workingSet;
+
+        public AppPoolListener(string appPoolName, string category, string counter)
+        {
+            this.appPoolName = appPoolName;
+            this.category = category;
+            this.counter = counter;
+
+            this.LoadCounterName();
+        }
 
         public AppPoolListener(string appPoolName)
         {
+            this.workingSet = true;
             this.appPoolName = appPoolName;
 
             this.LoadCounterName();
         }
 
-        public void LoadCounterName()
+        public bool LoadCounterName()
         {
             string newName = this.GetCounterName(this.appPoolName);
 
             if (!string.IsNullOrEmpty(newName) && this.counterName != newName)
             {
-                if (this.workingSetListener != null)
+                if (this.counterListener != null)
                 {
-                    this.workingSetListener.Dispose();
+                    this.counterListener.Dispose();
 
-                    this.workingSetListener = null;
+                    this.counterListener = null;
                 }
-
-                this.counterName = newName;
+				if (!string.IsNullOrEmpty(newName))
```

Comment:
Does this makes sense? I think the process for the app pool does not exist any more, if `newName` is `null` here. The `counterListener` variable is also already set to `null` therefore no value is reported.